### ISSUE TITLE
Tests fail with Node.js >= 4

### ISSUE THREAD

**Issue by hugo19941994** [Telefonica/node-jwt-utils#12]
One of the tests which builds a signed JWT with RS256 fails:

```
    1) should generate and read a JWT signed using RS256, privKey 1024
```

```
  1) Jwt Utils Tests should generate and read a JWT signed using RS256, privKey 1024:
     AssertionError: expected [Error: error:0607A082:digital envelope routines:EVP_CIPHER_CTX_set_key_length:invalid key length] to not exist
      at test/unit/jwt-utils-test.js:562:25
      at Object.buildJWT (lib/jwt-utils.js:425:16)
      at Context.<anonymous> (test/unit/jwt-utils-test.js:561:14)
```

Tested with Node.js 4, 6 and 8. iojs 2 works fine though

I will try to find what causes this

**Comment by palmerabollo** [comment]
I can confirm the issue.

**Comment by hugo19941994** [comment]
This is caused by a bug in NodeJS which was fixed from version 10.7 onwards (https://github.com/nodejs/node/pull/21288). I'm not sure if a backport of the fix to version 8.x is planned.

It looks like errors in OpenSSL's CipherBase queue were not being handled properly. This AssertionError is actually an error in the queue from a previous unit test which was supposed to fail and was not being cleaned up internally. If all unit tests except this one are removed it will actually work.

Not much we can do from here. Closing the issue.
